marlowe-playground-client/src/MainFrame.purs

```diff
@@ -316,6 +338,24 @@ handleAction s (HandleActusBlocklyMessage (ActusBlockly.CurrentTerms flavour ter
         Failure e -> void $ query _actusBlocklySlot unit (ActusBlockly.SetError ("Server error! " <> (showErrorDescription (runAjaxError e).description)) unit)
         _ -> void $ query _actusBlocklySlot unit (ActusBlockly.SetError "Unknown server error!" unit)
 
+handleAction s (ProjectsAction action@(Projects.LoadProject lang gistId)) = do
+  eGist <- flip runReaderT s $ runExceptT $ getApiGistsByGistId gistId
+  case eGist of
+    Right gist -> do
+      res <- toSimulation $ Simulation.loadGist gist
+      case res of
+        Right _ -> pure unit
+        Left error -> assign (_projects <<< Projects._projects) (Failure "Failed to load gist")
+    Left error -> assign (_projects <<< Projects._projects) (Failure "Failed to load gist")
+  toProjects $ Projects.handleAction s action
```

Comment:
The walking cases are usually a sign you should be using another monad.

The monad in this case would be `MonadError`, and you're in luck, you're already using it! You're just calling `runExceptT` too early.

Reply:
slightly improved

Reply:
:-P